### PR TITLE
Move utils/pooling to training/pooling.py

### PR DESCRIPTION
**Pull request by thomasjpfan**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [x] other (Refactor utils directory)

Addresses https://github.com/pytorch/torchtune/issues/1436

#### Changelog
This PR moves the `utils/pooling.py` to `training/pooling.py`. There is no API docs for pooling in `api_ref_utilities.rst `, so I did not make an update to `api_ref_training.rst`. 

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.)

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Example of docstring: https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285
Example in our docs: https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models

- [ ] I did not change any public API;
- [ ] I have added an example to docs or docstrings;

### CODE REVIEW

**Comment by RdoubleA** [comment]
Really appreciate you doing this. I actually ended up covering this in #1455 which I will merge. I'll be sure to review your other PR. Thanks for the contribution!

**Comment by ebsmothers** [comment]
Thanks for the PR! I am gonna close since this has already been handled in #1455
